e2e: TestKuke_Init_VerifyState tries to pull a non-existent image when run from a dirty tree

## Repro

On `origin/main` (462f83a) from any working tree that produces a `-dirty` `git describe` output (i.e., the test was run during local development with uncommitted edits):

```bash
make kuke           # builds kuke from a dirty tree → version baked in is e.g. v0.2.4-1-g462f83a-dirty
sudo env "PATH=/usr/local/go/bin:$PATH" HOME=$HOME make e2e
```

## Observed

```
=== NAME  TestKuke_Init_VerifyState
    e2e_kuke_test.go:248: running /home/.../kuke [--run-path tmp/e-... init] failed: exit status 1
        output:
        Creating space network 'default-default'
        Creating space network 'kuke-system-kukeon'
        time="…" level=info msg="fetch failed after status: 404 Not Found" host=ghcr.io
        Error: failed to create cell: failed to create container kukeon_kukeon_kukeond_kukeond:
          failed to create container from spec: failed to pull image
          ghcr.io/eminwux/kukeon:v0.2.4-1-g462f83a-dirty:
          failed to resolve reference "ghcr.io/eminwux/kukeon:v0.2.4-1-g462f83a-dirty":
          ghcr.io/eminwux/kukeon:v0.2.4-1-g462f83a-dirty: not found
--- FAIL: TestKuke_Init_VerifyState (1.78s)
```

Reproduces deterministically — anything that yields a version tag absent from the registry trips it.

## Root cause

`e2e/e2e_kuke_test.go:247` runs `kuke init` without passing `--kukeond-image`. The controller falls back to the default repo (`ghcr.io/eminwux/kukeon`) at the version baked into the binary by `Makefile:25-27`:

```
KUKEON_VERSION = $(shell git describe --tags --always --dirty --match 'v*')
LDFLAGS := … -X $(MODULE)/cmd/config.Version=$(KUKEON_VERSION) …
```

`git describe … --dirty` appends `-dirty` for any uncommitted edit, so the test resolves a tag the registry never received. The same will happen on any commit-since-tag (e.g. `v0.2.4-1-g462f83a`) until that exact SHA is published, which never happens for in-progress work.

The smoke-test path in CLAUDE.md sidesteps this by always passing `--kukeond-image docker.io/library/kukeon-local:dev`, so the failure only surfaces in the e2e suite.

## Suggested fix

Make the test pass an image that's guaranteed to exist locally. Options, in increasing order of test-rewrite cost:

1. Have the e2e harness build a deterministic-tagged image (e.g. `kukeon-e2e:test`) once per `go test ./e2e` invocation and pass `--kukeond-image kukeon-e2e:test` from `TestKuke_Init_VerifyState`. Mirrors what the Makefile smoke test already does manually.
2. Skip the test (or drop the kukeond cell assertion specifically) when `--kukeond-image` would resolve to a non-existent registry tag. Detect via `containerd images ls` or by short-circuiting on a `t.Skip` when `os.Getenv("KUKEOND_IMAGE")` is unset.
3. Add a Makefile target `e2e-with-image` that builds, loads, and exports `KUKEOND_IMAGE`, and have `TestKuke_Init_VerifyState` honor it (skip otherwise).

Option 1 keeps coverage; the others trade coverage for stability.

--- a/Makefile
+++ b/Makefile
@@ -68,11 +68,24 @@ kill:
 test:
 	go test $(shell go list ./... | grep -v /e2e)
 
+# Tag the e2e suite uses to refer to the local kukeond image. The e2e harness
+# imports it into containerd's kuke-system namespace before running `kuke init`,
+# so the test does not depend on a published registry tag matching `git
+# describe`.
+KUKEON_E2E_IMAGE_DOCKER_NAME ?= kukeon-local:e2e
+KUKEON_E2E_IMAGE ?= docker.io/library/$(KUKEON_E2E_IMAGE_DOCKER_NAME)
+
 e2e: test-e2e
 .PHONY: test-e2e
 test-e2e: kuke
+	@echo "Building local kukeond image $(KUKEON_E2E_IMAGE_DOCKER_NAME) for e2e"
+	docker build --build-arg VERSION=v0.0.0-e2e -t $(KUKEON_E2E_IMAGE_DOCKER_NAME) .
 	@echo "Running e2e tests using binaries in project root"
-	HOME=$(HOME) PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:$(PATH) E2E_BIN_DIR=$(CURDIR) go test -v ./e2e
+	HOME=$(HOME) PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:$(PATH) \
+		E2E_BIN_DIR=$(CURDIR) \
+		KUKEON_E2E_IMAGE=$(KUKEON_E2E_IMAGE) \
+		KUKEON_E2E_IMAGE_DOCKER_NAME=$(KUKEON_E2E_IMAGE_DOCKER_NAME) \
+		go test -v ./e2e
 
 tag:
 	git tag -s v$(KUKEON_VERSION) -m "Release version $(KUKEON_VERSION)"

--- a/e2e/e2e_kuke_test.go
+++ b/e2e/e2e_kuke_test.go
@@ -243,8 +243,14 @@ func TestKuke_Init_VerifyState(t *testing.T) {
 		cleanupRealm(t, runPath, "default")
 	})
 
+	// Step 0.5: Stage the local kukeond image into containerd's kuke-system
+	// namespace. The Step 0 purge above wipes any prior image, and dev builds
+	// from a dirty/untagged tree resolve to a registry tag that doesn't exist,
+	// so init would fail on image pull without this.
+	kukeondImage := loadKukeondImageIntoContainerd(t)
+
 	// Step 1: Run init command
-	args := append(buildKukeRunPathArgs(runPath), "init")
+	args := append(buildKukeRunPathArgs(runPath), "init", "--kukeond-image", kukeondImage)
 	output := runReturningBinary(t, nil, kuke, args...)
 
 	// Step 2: Verify bootstrap report output contains expected content

--- a/e2e/e2e_test.go
+++ b/e2e/e2e_test.go
@@ -157,6 +157,66 @@ func buildKukeRunPathArgs(runPath string) []string {
 	return []string{"--run-path", runPath}
 }
 
+// loadKukeondImageIntoContainerd stages the local kukeond image into the
+// kuke-system containerd namespace and returns the fully-qualified image
+// reference to pass via `kuke init --kukeond-image`. It skips the test when
+// the harness env vars are unset (running `go test ./e2e` without the
+// `make e2e` wrapper) or when `docker` / `ctr` aren't on PATH. `kuke init`
+// otherwise resolves the kukeond image from `git describe` output, which
+// fails on dirty/untagged dev trees because that ref does not exist in any
+// registry.
+func loadKukeondImageIntoContainerd(t *testing.T) string {
+	t.Helper()
+
+	image := os.Getenv("KUKEON_E2E_IMAGE")
+	dockerName := os.Getenv("KUKEON_E2E_IMAGE_DOCKER_NAME")
+	if image == "" || dockerName == "" {
+		t.Skip(
+			"KUKEON_E2E_IMAGE / KUKEON_E2E_IMAGE_DOCKER_NAME unset; " +
+				"run via 'make e2e' (which builds the local kukeond image) " +
+				"or set both env vars manually",
+		)
+	}
+
+	dockerPath, err := exec.LookPath("docker")
+	if err != nil {
+		t.Skipf("docker binary not found, skipping kukeond image staging: %v", err)
+	}
+	ctrPath, err := exec.LookPath(ctr)
+	if err != nil {
+		t.Skipf("ctr binary not found, skipping kukeond image staging: %v", err)
+	}
+
+	tarPath := filepath.Join(t.TempDir(), "kukeond.tar")
+
+	saveCtx, saveCancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer saveCancel()
+	saveCmd := exec.CommandContext(saveCtx, dockerPath, "save", "-o", tarPath, dockerName)
+	if out, saveErr := saveCmd.CombinedOutput(); saveErr != nil {
+		t.Fatalf(
+			"docker save %q failed: %v\noutput:\n%s\n"+
+				"hint: 'make e2e' builds this image; if running go test directly, build it first",
+			dockerName, saveErr, string(out),
+		)
+	}
+
+	importCtx, importCancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer importCancel()
+	importCmd := exec.CommandContext(
+		importCtx, ctrPath,
+		"--namespace", consts.KukeSystemRealmNamespace,
+		"images", "import", tarPath,
+	)
+	if out, importErr := importCmd.CombinedOutput(); importErr != nil {
+		t.Fatalf(
+			"ctr -n %s images import %q failed: %v\noutput:\n%s",
+			consts.KukeSystemRealmNamespace, tarPath, importErr, string(out),
+		)
+	}
+
+	return image
+}
+
 // verifyContainerdNamespace verifies containerd namespace exists by running ctr ns ls.
 func verifyContainerdNamespace(t *testing.T, namespace string) bool {
 	t.Helper()

